Client/Frontend/Browser/New Tab Page/Sections/StatsSectionProvider.swift

```diff
@@ -118,7 +118,20 @@ class BraveShieldStatsView: UIView, Themeable {
             $0.allowedUnits = [.useKB, .useMB, .useGB, .useTB]
         }
         
-        return formatter.string(fromByteCount: Int64(estimatedDataSavedInBytes))
+        // Byte formatted megabytes value can be too long to display nicely(#3274).
+        // As a workaround we trim the decimal value when megabytes or kilobytes are displayed.
+        // For gigabytes it takes a lot of time to make the stat 100GB+ to be a problem.
+        // In such case the stat font is being scaled down.
+        // In the future we might have to write our own byte formater to handle this.
+        let byteFormattedString = formatter.string(fromByteCount: Int64(estimatedDataSavedInBytes))
+        let _1MB = 1000 * 1000
+        if estimatedDataSavedInBytes / _1MB < 1000 {
+            if let withRemovedDecimal = try? byteFormattedString.regexReplacePattern("\\.[0-9]+", with: "") {
+                return withRemovedDecimal
+            }
+        }
```

Comment:
Also we should only be doing this for 1MB+, `ByteCountFormatter` does not display any fractional digits for KB and lower

Reply:
how to get localized unit value then? as on attached screenshot It's `1MB` in most locales, but in french for example it says `1Mo` 
Perhaps I should just drop the byte formatter altogether and make custom logic for it, and force english locale(no locale property in byteCountFormatter afaik)

Reply:
Localization doesn't matter since we're talking about numbers. A megabyte in english and french are both 1 million bytes and it will still only show 1 fractional digit for MB and 2 fractional digits for GB (as documented under [`ByteCountFormatter.isAdaptive`](https://developer.apple.com/documentation/foundation/bytecountformatter/1417887-isadaptive)) regardless of language.

Reply:
oh I could round up before formatting maybe

Reply:
Yeah, altered logic would be checking if `estimatedDataSavedInBytes` is > 1MB, and if so, rounding it via `(estimatedDataSavedInBytes / _1MB) * _1MB`, then passing that value into the formatter